compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/BIRPackageSymbolEnter.java

```diff
@@ -632,6 +632,14 @@ private void addShapeCP(BType bType, int typeCpIndex) {
     }
 
     private void defineAnnotations(DataInputStream dataInStream) throws IOException {
+        BAnnotationSymbol annotationSymbol = defineAnnotation(dataInStream);
+        this.env.pkgSymbol.scope.define(annotationSymbol.name, annotationSymbol);
```

Comment:
Isn't this problematic? This defines the annotation in the currently defining module's scope. Wouldn't this be called for annotations from outside modules as well? 🤔

Reply:
This is actually defining the symbols for the annotation definitions in the same module (previously https://github.com/ballerina-platform/ballerina-lang/blob/master/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/BIRPackageSymbolEnter.java#L634). I just extracted out the `BAnnotationSymbol` reading logic so that it can be reused when reading the attachment symbols also. There we just add them to the symbol (https://github.com/ballerina-platform/ballerina-lang/pull/34329/files#diff-75a4933bf7e446d096c3a9ce1c8e8abb41cd5497adba2c1a9c84093240169519R874).